linux/LinuxProcess.c

```diff
@@ -381,180 +402,139 @@ void LinuxProcess_makeCommandStr(Process* this) {
-         cmdline += matchLen;
-
-         if (commInCmdline) {
-            commStart += str - strStart - matchLen;
-            commEnd += str - strStart - matchLen;
-         }
-      } else {
-         /* cmdline will be a separate field */
-         mc->sep1 = str - strStart;
-         str = stpcpy(str, SEPARATOR);
-
-         if (commInCmdline) {
-            commStart += str - strStart - SEPARATOR_LEN + 1;
-            commEnd += str - strStart - SEPARATOR_LEN + 1;
-         }
-      }
-
-      mc->separateComm = false;  /* procComm merged */
-   } else {
-      mc->sep1 = str - strStart;
-      str = stpcpy(str, SEPARATOR);
+   bool haveCommField = false;
 
-      commStart = str - strStart - SEPARATOR_LEN + 1;
+   if (!haveCommInExe && !haveCommInCmdline && procComm) {
+      WRITE_SEPARATOR;
+      WRITE_HIGHLIGHT(0, strlen(procComm), commAttr, CMDLINE_HIGHLIGHT_FLAG_COMM);
       str = stpcpy(str, procComm);
-      commEnd = str - strStart - SEPARATOR_LEN + 1;   /* or commStart + strlen(procComm) */
+      haveCommField = true;
+   }
 
-      mc->unmatchedExe = !matchLen;
+   if (matchLen) {
+      /* strip the matched exe prefix */
+      cmdline += matchLen;
 
-      if (matchLen) {
-         if (stripExeFromCmdline) {
-            cmdline += matchLen;
-         }
-      }
-
-      if (*cmdline) {
-         mc->sep2 = str - strStart - SEPARATOR_LEN + 1;
-         str = stpcpy(str, SEPARATOR);
-      }
+      commStart -= matchLen;
+      commEnd -= matchLen;
+   }
 
-      mc->separateComm = true;  /* procComm a separate field */
+   if (!matchLen || (haveCommField && *cmdline)) {
+      /* cmdline will be a separate field */
+      WRITE_SEPARATOR;
    }
 
+   if (!haveCommInExe && haveCommInCmdline && !haveCommField)
+      WRITE_HIGHLIGHT(commStart, commEnd - commStart, commAttr, CMDLINE_HIGHLIGHT_FLAG_COMM);
+
    /* Display cmdline if it hasn't been consumed by procExe */
    if (*cmdline) {
       (void) stpcpyWithNewlineConversion(str, cmdline);
    }
 
-   mc->commStart = commStart;
-   mc->commEnd = commEnd;
+   #undef WRITE_SEPARATOR
+   #undef WRITE_HIGHLIGHT
 }
 
 static void LinuxProcess_writeCommand(const Process* this, int attr, int baseAttr, RichString* str) {
+   (void)baseAttr;
+
    const LinuxProcess *lp = (const LinuxProcess *)this;
    const LinuxProcessMergedCommand *mc = &lp->mergedCommand;
 
    int strStart = RichString_size(str);
 
-   int baseStart = strStart + lp->mergedCommand.baseStart;
-   int baseEnd = strStart + lp->mergedCommand.baseEnd;
-   int commStart = strStart + lp->mergedCommand.commStart;
-   int commEnd = strStart + lp->mergedCommand.commEnd;
+//   int commAttr = CRT_colors[Process_isUserlandThread(this) ? PROCESS_THREAD_COMM : PROCESS_COMM];
 
-   int commAttr = CRT_colors[Process_isUserlandThread(this) ? PROCESS_THREAD_COMM : PROCESS_COMM];
+   const bool highlightBaseName = this->settings->highlightBaseName;
+   const bool highlightSeparator = true;
```

Comment:
why a variable that is always `true`?

Reply:
To have a reference for later, which places are affected when highlighting a separator. Will iron out once the overall code structure has been better streamlined overall.

Reply:
fair enough